ocfl-java-core/src/main/java/edu/wisc/library/ocfl/core/extension/storage/layout/FlatOmitPrefixLayoutExtension.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2019 University of Wisconsin Board of Regents
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package edu.wisc.library.ocfl.core.extension.storage.layout;
+
+import edu.wisc.library.ocfl.api.OcflConstants;
+import edu.wisc.library.ocfl.api.exception.OcflExtensionException;
+import edu.wisc.library.ocfl.core.extension.OcflExtensionConfig;
+import edu.wisc.library.ocfl.core.extension.storage.layout.config.FlatOmitPrefixLayoutConfig;
+
+/**
+ * Implementation of the <a href="https://ocfl.github.io/extensions/0006-flat-omit-prefix-storage-layout.html">
+ * Flat Omit Prefix Storage Layout</a> extension.
+ *
+ * @author awoods
+ * @since 2021-06-22
+ */
+public class FlatOmitPrefixLayoutExtension implements OcflStorageLayoutExtension {
+
+    public static final String EXTENSION_NAME = "0006-flat-omit-prefix-storage-layout";
+
+    private FlatOmitPrefixLayoutConfig config;
+    private boolean isCaseSensitive = true; // The spec does not provide for configuring this
```

Comment:
During the `init` call you could do something like:

```java
isCaseSensitive = !delimiter.toLowerCase().equals(delimiter.toUpperCase());
```

Then, if it's not case sensitive, there's no need to lower case the object id on `mapObjectId`.

If you don't care about this optimization, then we probably don't need the `isCaseSensitive` field.